beacon-chain/blockchain/core.go

```diff
@@ -222,18 +222,22 @@ func (b *BeaconChain) verifyBlockTimeStamp(block *types.Block) (bool, error) {
 // computeNewActiveState for every newly processed beacon block.
 func (b *BeaconChain) computeNewActiveState(attestations []*pb.AttestationRecord, activeState *types.ActiveState, blockVoteCache map[[32]byte]*types.VoteCache, blockHash [32]byte) (*types.ActiveState, error) {
 	// TODO: Insert recent block hash.
-	activeState.SetBlockVoteCache(blockVoteCache)
-	activeState.NewPendingAttestation(attestations)
-	blockHashes := activeState.RecentBlockHashes()
-	blockHashes = append(blockHashes, blockHash)
-
-	for len(blockHashes) > 2*params.CycleLength {
-		blockHashes = blockHashes[1:]
+	var recentBlockHashes [][]byte
+	for _, blockHashes := range activeState.RecentBlockHashes() {
+		recentBlockHashes = append(recentBlockHashes, blockHashes[:])
+	}
+	recentBlockHashes = append(recentBlockHashes, blockHash[:])
+	for len(recentBlockHashes) > 2*params.CycleLength {
```

Comment:
This is checking for `recentblockHashes` should only has the most recent `2 * CYCLE_LENGTH block hashes, older to newer`

I think is better if do this check inside `ReplaceBlockHashes` function

Reply:
I'm not using the ReplaceBlockHashs function here since it altogether is creating a new object, should I make this change? (Which would mean creating a activeState variable and replacing the block hashes for that)